Fix - Go Back buttons in options menu

IgnitedIce opened the pull request:
Made a new pull request, this time without the mess

**Original PR:**
Fixed issue #2031 
Changed "Go Back" buttons in options to open options menu instead of main menu.

maxsupermanhd commented:
You could just force push changes to the branch btw

IgnitedIce commented:
> You could just force push changes to the branch btw

yeah git is not rly my strong side :(
anyway, is it also a valid option?

maxsupermanhd commented:
Weird one but it works

KJeff01 commented on src/ingameop.cpp:
```diff
@@ -183,7 +183,7 @@ static bool addAudioOptions()
 #endif
 
-	addIGTextButton(INTINGAMEOP_GO_BACK, INTINGAMEOP_1_X, INTINGAMEOPAUTO_Y_LINE(row), INTINGAMEOP_SW_W, _("Go Back"), OPALIGN);
+	addIGTextButton(INTINGAMEOP_AUDIO_BACK, INTINGAMEOP_1_X, INTINGAMEOPAUTO_Y_LINE(row), INTINGAMEOP_SW_W, _("Go Back"), OPALIGN);
```
I'm sure it works fine but this shouldn't be necessary. A "isAudioOptionsUp" flag with a similar "runIGAudioOptionsMenu" function would be more in line with how the rest of the in-game menus operate.

IgnitedIce replied:
I agree, I rushed this fix because I don't have access to my main pc rn so I wanted to push this asap, anyway I'd rather hold this pr until I can rework this code, it works but I'm not rly happy with my solution..

KJeff01 commented on src/ingameop.cpp:
```diff
@@ -611,6 +593,30 @@ static bool startIGOptionsMenu()
 
+void intReopenMenuWithoutUnPausing(bool toOptionsMenu)
+{
+	if (toOptionsMenu)
+	{
+		startIGOptionsMenu();
+		return;
+	}
```
I'd prefer if you didn't move this entire function down a few lines. That way the diff can be looked over faster as to what changed.

IgnitedIce replied:
yeah I moved the function below because I added a call to another function which was below that one so it was unreachable

KJeff01 replied:
True. I didn't notice that, heh. :+1:

IgnitedIce commented:
I rewrote my code to match the review, feedback appreciated.

KJeff01 commented:
:+1: Close but it can be better (the wz source can be such a pain lol). I was going to tackle this issue myself months ago but never got to it.

So in `intProcessInGameOptions()` there are a bunch of if statements with the flag for the respective option at the top. You'll want to add something similar to this, instead:

```
else if (isAudioOptionsUp)
{
	if (runIGAudioOptionsMenu(id))
	{
		intCloseInGameOptions(true, true);
	}
	return;
}
```

This new "run" function will handle the audio options menu button events rather than those in `intProcessInGameOptions()` (and you can just take those out from the switch case in that function into the new run function). IDK why the audio menu doesn't have a special function to handle this itself... but we should fix that finally.

You can take a look at an existing run function like: `runIGMouseOptionsMenu() ` for your new "runIGAudioOptionsMenu()".

This way you won't have to mess with replacing all those `intReopenMenuWithoutUnPausing()` with `startIGOptionsMenu()` which currently does a behavior change for the network host. Or adding a specific exit option to a `INTINGAMEOP_GO_BACK` case just for the audio menu.

IgnitedIce commented:
Alright, will do. 
yeah I was confused about the audio options.. 😅

IgnitedIce commented:
Whoops, tried renaming the branch.
Anyway, these are the changes I've made:
Added the `runIGAudioOptionsMenu` function; renamed `addAudioOptions` to `startIGAudioOptionsMenu` and moved it down; moved the audio option cases from `intProcessInGameOptions` to the new audio 'run' function.

KJeff01 commented:
Yes, looks nice now! Appears it is working from my tests. I'll keep this open for others to try or review for a bit.
